tests/staking/observable/createStakingTx.test.ts

```diff
@@ -0,0 +1,164 @@
+    };
+    expect(() => new ObservableStaking(network, stakerInfoWithCoordPk)).toThrow(
+      "Invalid staker public key"
+    );
+
+    const stakerInfoWithInvalidAddress = {
+      address: "abc",
+      publicKeyNoCoordHex: stakerInfo.publicKeyNoCoordHex,
+    };
+    expect(() => new ObservableStaking(network, stakerInfoWithInvalidAddress)).toThrow(
+      "Invalid staker bitcoin address"
+    );
+  });
+
+  it(`${networkName} should throw an error if input data validaiton failed`, async () => {
+    jest.spyOn(stakingUtils, "validateStakingTxInputData").mockImplementation(() => {
+      throw new StakingError(StakingErrorCode.INVALID_INPUT, "some error");
+    });
+    const observableStaking = new ObservableStaking(network, stakerInfo);
+
+    expect(() => observableStaking.createStakingTransaction(
+      params,
+      params.minStakingAmountSat,
+      stakingTerm,
+      finalityProviderPublicKey,
+      utxos,
+      feeRate,
+    )).toThrow(
+      new StakingError(StakingErrorCode.INVALID_INPUT, "some error")
+    );
+  });
+
+  it(`${networkName} should throw an error if fail to build scripts`, async () => {
+    jest.spyOn(stakingScript, "StakingScriptData").mockImplementation(() => {
+      throw new StakingError(StakingErrorCode.SCRIPT_FAILURE, "some error");
+    });
+    const observableStaking = new ObservableStaking(network, stakerInfo);
+
+    expect(() => observableStaking.createStakingTransaction(
+      params,
+      params.minStakingAmountSat,
+      stakingTerm,
+      finalityProviderPublicKey,
+      utxos,
+      feeRate,
+    )).toThrow(
+      new StakingError(StakingErrorCode.SCRIPT_FAILURE, "some error")
+    );
+  });
+
+  it(`${networkName} should throw an error if fail to build staking tx`, async () => {
+    jest.spyOn(staking, "stakingTransaction").mockImplementation(() => {
+      throw new Error("fail to build staking tx");
+    });
+    const observableStaking = new ObservableStaking(network, stakerInfo);
+
+    expect(() => observableStaking.createStakingTransaction(
+      params,
+      params.minStakingAmountSat,
+      stakingTerm,
+      finalityProviderPublicKey,
+      utxos,
+      feeRate,
+    )).toThrow(
+      new StakingError(StakingErrorCode.BUILD_TRANSACTION_FAILURE, "fail to build staking tx")
+    );
+  });
+
+  it(`${networkName} should successfully create a staking transaction`, async () => {
+    const observableStaking = new ObservableStaking(network, stakerInfo);
+    const amount = dataGenerator.getRandomIntegerBetween(
+      params.minStakingAmountSat, params.maxStakingAmountSat,
+    );
+    const { psbt, fee} = observableStaking.createStakingTransaction(
+      params,
+      amount,
+      stakingTerm,
+      finalityProviderPublicKey,
+      utxos,
+      feeRate,
+    );
+
+    expect(psbt).toBeDefined();
+    expect(fee).toBeGreaterThan(0);
+    
+    // Check the inputs
+    expect(psbt.data.inputs.length).toBe(1);
```

Comment:
Why would this always be `1`? The staking amount might require more utxos to be satisfied